source/dol/bases/d_CourseSelectGuide.cpp

```diff
@@ -0,0 +1,1000 @@
+STATE_DEFINE(dCourseSelectGuide_c, RightOnStageAnimeEndCheck);
+STATE_DEFINE(dCourseSelectGuide_c, RightDisp);
+STATE_DEFINE(dCourseSelectGuide_c, RightExitAnimeEndCheck);
+STATE_DEFINE(dCourseSelectGuide_c, ShadowOnStageWait);
+STATE_DEFINE(dCourseSelectGuide_c, ShadowOnStageAnimeEndCheck);
+STATE_DEFINE(dCourseSelectGuide_c, ShadowDisp);
+STATE_DEFINE(dCourseSelectGuide_c, ShadowExitAnimeEndCheck);
+STATE_DEFINE(dCourseSelectGuide_c, ScrollGuideOnStageWait);
+STATE_DEFINE(dCourseSelectGuide_c, ScrollGuideOnStageAnimeEndCheck);
+STATE_DEFINE(dCourseSelectGuide_c, ScrollGuideDisp);
+STATE_DEFINE(dCourseSelectGuide_c, ScrollGuideExitAnimeEndCheck);
+
+dCourseSelectGuide_c::~dCourseSelectGuide_c() {
+    m_instance = nullptr;
+}
+
+bool dCourseSelectGuide_c::createLayout() {
+    static const char *AnmNameTbl[ANIM_NAME_COUNT] = {
+        "corseSelectUIGuide_37_inInfo.brlan",
+        "corseSelectUIGuide_37_loopInfo.brlan",
+        "corseSelectUIGuide_37_outInfo.brlan",
+        "corseSelectUIGuide_37_inMapArrow.brlan",
+        "corseSelectUIGuide_37_loopArrow.brlan",
+        "corseSelectUIGuide_37_outMapArrow.brlan",
+        "corseSelectUIGuide_37_inShadow.brlan",
+        "corseSelectUIGuide_37_outShadow.brlan"
+    };
+
+    static const char *GROUP_NAME_DT[ANIM_COUNT] = {
+        "A00_underGuide", "A01_zanki", "A02_courseInfo",
+        "A00_underGuide", "A01_zanki", "A02_courseInfo",
+        "A00_underGuide", "A01_zanki", "A02_courseInfo",
+        "B00_ArrowRight", "B01_ArrowLeft", "B02_ArrowUp", "B03_ArrowUnder",
+        "B00_ArrowRight", "B01_ArrowLeft", "B02_ArrowUp", "B03_ArrowUnder",
+        "B00_ArrowRight", "B01_ArrowLeft", "B02_ArrowUp", "B03_ArrowUnder",
+        "C00_shadow",
+        "C00_shadow"
+    };
+
+    static const int ANIME_INDEX_TBL[ANIM_COUNT] = {
+        inInfo, inInfo, inInfo,
+        loopInfo, loopInfo, loopInfo,
+        outInfo, outInfo, outInfo,
+        inMapArrow, inMapArrow, inMapArrow, inMapArrow,
+        loopArrow, loopArrow, loopArrow, loopArrow,
+        outMapArrow, outMapArrow, outMapArrow, outMapArrow,
+        inShadow,
+        outShadow
+    };
+
+    static const char *NPANE_NAME_DT[N_COUNT] = {
+        "N_IconPos1P_00", "N_IconPos2P_00", "N_IconPos3P_00", "N_IconPos4P_00",
+        "N_mapArrow_00",
+        "N_proportionL_00", "N_proportionR_00", "N_proportionC_00",
+        "N_guideViewC_00", "N_guideViewR_01",
+        "N_left_00"
+    };
+
+    static const char *T_PANE_NAME_TBL[T_COUNT] = {
+        "T_worldNum_00",
+        "T_cSelect_00", "T_cSelect_pic",
+        "T_lifeNumber_00", "T_lifeNumber_01", "T_lifeNumber_02", "T_lifeNumber_03",
+        "T_guideViewLS_00", "T_guideViewL_01"
+    };
+
+    static const int MESSAGE_DATA_TBL[] = {
+        MSG_CS_SELECT_WORLD,
+        MSG_CS_MENU,
+        MSG_CS_WORLD
+    };
+    static const char *T_PANE_FIXED_NAME_TBL[] = {
+        "T_guideViewC_00",
+        "T_guideViewR_00",
+        "T_world_00"
+    };
```

Comment:
Please define a T_FIXED_PANE_e enum with the values fromT_PANE_FIXED_NAME_TBL and T_FIXED_COUNT. This allows us to ensure these two arrays have the same size.